nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -257,11 +302,22 @@ BtNavigator::navigateToPose()
       action_server_->publish_feedback(feedback_msg);
     };
 
+  auto bt_id = action_server_->get_current_goal()->behavior_tree;
+
+  // Empty id in request is default for backward compatibility
+  if (bt_id == "") {
+    bt_id = bt_xml_ids_[0];
+  }
+
+  if (!loadBehaviorTree(bt_id)) {
```

Comment:
You should check if this tree is already the same one already loaded so you don't load multiple times.

Reply:
Yes, it is done in the first sentence in `loadBehaviorTree`